ui/src/router.js

```diff
@@ -66,37 +129,129 @@ let router = new Router({
+          var allowed=checkPageRole(roles,routeName);
+          if(allowed){
+            next();
+          }
+          else{
+            var error = "The user does not have the necessary privileges to access the page : "+routeName;
+            next({
+              path: "/noaccess",
+            });
+            vueInstance.displayError(error);
+          }
+          
+        }
     },
     {
       path: "/people/add/:section/:id",
       name: "edit-people",
       component: () =>
-        import(/* webpackChunkName: "EditPeople" */ "./views/People/AddSubsection.vue")
+        import(/* webpackChunkName: "EditPeople" */ "./views/People/AddSubsection.vue"),
+        beforeEnter (to, from, next) {
+          var routeName=to.name;
+          var roles=store.state.roles;
+          var allowed=checkPageRole(roles,routeName);
+          if(allowed){
+            next();
+          }
+          else{
+            var error = "The user does not have the necessary privileges to access the page : "+routeName;
+            next({
+              path: "/noaccess",
+            });
+            vueInstance.displayError(error);
+          }
+        }
     },
     {
       path: "/people/edit/:id",
       name: "people-edit",
       component: () =>
-        import(/* webpackChunkName: "ViewPeople" */ "./views/People/Edit.vue")
+        import(/* webpackChunkName: "ViewPeople" */ "./views/People/Edit.vue"),
+        beforeEnter (to, from, next) {
+          var routeName=to.name;
+          var roles=store.state.roles;
+          var allowed=checkPageRole(roles,routeName);
+          if(allowed){
+            next();
+          }
+          else{
+            var error = "The user does not have the necessary privileges to access the page : "+routeName;
+            next({
+              path: "/noaccess",
+            });
+            vueInstance.displayError(error);
+          }
+        }
     },
     {
       path: "/people/view/:id",
       name: "people-view",
       component: () =>
-        import(/* webpackChunkName: "ViewPeople" */ "./views/People/View.vue")
+        import(/* webpackChunkName: "ViewPeople" */ "./views/People/View.vue"),
+        beforeEnter (to, from, next) {
+          var routeName=to.name;
+          var roles=store.state.roles;
+          var allowed=checkPageRole(roles,routeName);
+          if(allowed){
+            next();
+          }
+          else{
+            var error = "The user does not have the necessary privileges to access the page : "+routeName;
+            next({
+              path: "/noaccess",
+            });
+            vueInstance.displayError(error);
+          }
+        }
     },
     {
       path: "/relationship",
       name: "relationship",
       component: () =>
-        import(/* webpackChunkName: "about" */ "./views/Relationship/Relationship.vue")
+        import(/* webpackChunkName: "about" */ "./views/Relationship/Relationship.vue"),
+        beforeEnter (to, from, next) {
```

Comment:
Rather than adding this to each path, is there a way to include it as part of the router.beforeEnter? You could just check the page each time and if there is not an entry in the checkPageRoles function, then it assumes true. Like for the home page or login page for example.